src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -648,7 +648,7 @@ private void HandleFinished(bool success, ClientSideStatus receivedStatus)
             if (status.StatusCode != StatusCode.OK)
             {
                 streamingResponseCallFinishedTcs.SetException(new RpcException(status, receivedStatus.Trailers));
-                if (status.StatusCode == StatusCode.Cancelled || origCancelRequested)
+                if (status.StatusCode == StatusCode.Cancelled || status.StatusCode == StatusCode.Unavailable || origCancelRequested)
                 {
                     // Make sure the exception set to the Task is observed,
```

Comment:
we could potentially reuse the  Task.ObserveException utility method from here: https://github.com/grpc/grpc-dotnet/pull/1660/files#diff-cf6772e9387e5b8b33dd040aed96eccded75c5937df6af33f242396871f802f2R32